app/mutator.py

```diff
@@ -35,7 +35,10 @@ class Mutator(app.selectable.Selectable):
   """Track and enact changes to a body of text."""
   def __init__(self):
     app.selectable.Selectable.__init__(self)
-    self.__compoundChange = None
+    self.__compoundChange = []
+    # |oldRedoIndex| is used to store the redo index before an action occurs,
+    # so we know where to insert the compound change.
+    self.oldRedoIndex = 0
```

Comment:
how are oldRedoIndex and tempChange related (are they). They seem related, but I haven't pinned down there are two things that seem to be doing a similar thing (if they are similar).

Reply:
They are kind of related. oldRedoIndex stores the current position of the redoIndex before the new action is handled. This is because if an action contains 5 changes, we redo all 5 of them, then we go back and combine them into one and insert it at oldRedoIndex. However, if there was a tempChange, we have to insert it as its own separate change and add it to the redoChain and update oldRedoIndex by adding 1.